apis/notification/v1alpha1/snstopic_types.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// Tag represetnt a user-provided metadata that can be associated with a
+// SNS Topic. For more information about tagging,
+// see Tagging SNS Topics (https://docs.aws.amazon.com/sns/latest/dg/sns-tags.html)
+// in the SNS User Guide.
+type Tag struct {
+
+	// The key name that can be used to look up or retrieve the associated value.
+	// For example, Department or Cost Center are common choices.
+	Key string `json:"key"`
+
+	// The value associated with this tag. For example, tags with a key name of
+	// Department could have values such as Human Resources, Accounting, and Support.
+	// Tags with a key name of Cost Center might have values that consist of the
+	// number associated with the different cost centers in your company. Typically,
+	// many resources have tags with the same key name but with different values.
+	//
+	// AWS always interprets the tag Value as a single string. If you need to store
+	// an array, you can store comma-separated values in the string. However, you
+	// must interpret the value in your code.
+	// +optional
+	Value string `json:"value,omitempty"`
+}
+
+// SNSTopicParameters define the desired state of a AWS SNS Topic
+type SNSTopicParameters struct {
+	// Name refers to the name of the AWS SNS Topic
+	// +immutable
+	Name string `json:"name"`
```

Comment:
Could we use the external name annotation for this?

Reply:
So, this Name field represents the name of SNS Topic, and the external name is updated with the ARN of this SNS Topic. If topicARN is not needed as external-name, we can shift it to status.
Not fixing this as of now, if you agree with the changes, I can go ahead.

Reply:
Got it - upon deeper review the pattern you have now makes sense. Thanks!